### PR TITLE
Addition of AI App Reference Doc

### PR DESCRIPTION
**Pull request by mynamebrody**

### CODE REVIEW

**Comment by justin-lee-zapier** [comment]
When you're ready for @team-platform-support to review this, please head over to this [Slack thread](https://zapier.slack.com/archives/C02PCUNBUG7/p1717101686230569) and give us a `:review-pr:` emoji.

File: docs/_reference/use-cases/ai-app.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+AI app integrations built on Zapier allow users to automate tasks using AI capabilities. Here are some common pain points and recommendations when building AI apps on Zapier.
+
+## Pain Points
+### 1. Long-Running Actions  
+
+**Pain Point:** Actions that take a long time to process can cause issues, especially with Zapier's execution time limits. Test steps in the Zapier editor are limited to 50 seconds, and live Zap executions default to 45 seconds.
+
+**Recommendations:**
+
+* Use `z.generateCallbackUrl()` and `performResume`
```
Comment (marinahand) [inline docs/_reference/use-cases/ai-app.md:20]: add link to https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#zgeneratecallbackurl

File: docs/_reference/use-cases/ai-app.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  * If you need more time for live Zap executions, you can request an extension by contacting Zapier Partner Support through their [contact form](https://zapier.com/partner/contact/). However, we can only do this so much and the most foolproof way would be to implement the previous recommendation.
+  * **Why:** Extending the execution time for live Zaps can help accommodate actions that inherently take longer to process.
+
+### 2. Handling Samples for Long-Running Tasks  
+
+**Pain Point:** Generating accurate samples for long-running tasks can be tricky, especially during the initial setup.
+
+**Recommendation:**
+
+* Use `bundle.meta.isLoadingSample`:
```
Comment (marinahand) [inline docs/_reference/use-cases/ai-app.md:34]: add link to https://github.com/zapier/zapier-platform/blob/main/packages/cli/README.md#bundlemeta

File: docs/_reference/use-cases/ai-app.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  * **Why:** This approach ensures that users get a quick response during setup, avoiding delays caused by the actual long-running task.
+
+### 3. Hiding Complex Fields in Actions  
+
+**Pain Point:** Complex configuration fields can overwhelm users, making the setup process cumbersome.
+
+**Recommendation:**
+
+* Use Custom Input Fields:  
+  * You can create an input field that depends on another. For example, you could create an “Advanced Features” input field at the bottom of all of your other ones that has the property `altersDynamicFields: true` so when it’s updated, all of the other fields refresh. Then for any some additional input fields, they would be dependent if that “Advanced Features” input field is true, then display the more advanced ones that are not necessarily required. (ie. `if (bundle.inputData.advanced === true)`)
```
Comment (marinahand) [inline docs/_reference/use-cases/ai-app.md:45]: include a link to https://github.com/zapier/zapier-platform/blob/main/packages/schema/docs/build/schema.md#fieldschema

Comment (marinahand) [inline docs/_reference/use-cases/ai-app.md:45]: suggestion:
"Then any additional input fields could be dependent on whether that “Advanced Features” input field is true, then display the more advanced ones that are not necessarily required for all users."

File: docs/_reference/use-cases/ai-app.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  * For example, a “Send Prompt" action could have a dropdown of available “use-cases” that when selected, pre-fill input fields for the user. This can be accomplished by having a mapping of sorts for your use-cases and some complex logic using `altersDynamicFields: true` to then have all of your input fields as custom ones which are loaded in via functions based on the the use-case selected and the default of them pre-filled.
+  * **Why:** Pre-configured AI prompts tailored to specific use cases streamline the user experience, making it easier for users to set up and use your app effectively and for them to understand how the AI is working where they could then tweak the prompt/other input fields themselves if need be.
+
+### 5. Value of Zap Templates  
+
+**Pain Point:** Users often need help figuring out how to set up Zaps for common use cases.
+
+**Recommendation:**
+
+* Provide Zap Templates:
```
Comment (marinahand) [inline docs/_reference/use-cases/ai-app.md:68]: link to https://platform.zapier.com/publish/zap-templates

File: docs/_reference/use-cases/ai-app.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+**Pain Point:** Users may struggle with setting up generalized actions that require specific configurations.
+
+**Recommendations:**
+
+* Create Use-Case Specific Actions:  
+  * For example, a “ChatGPT Summarize Text" action can hide the configuration and prompt details behind the scenes. This can be accomplished by hardcoding a prompt into the action configuration itself and then just have the user input the text that needs to be summarized.
+  * Why: Pre-configured actions tailored to specific use cases streamline the user experience, making it easier for users to set up and use your app effectively.
+* Create Use-Case Specific Dropdown:  
+  * For example, a “Send Prompt" action could have a dropdown of available “use-cases” that when selected, pre-fill input fields for the user. This can be accomplished by having a mapping of sorts for your use-cases and some complex logic using `altersDynamicFields: true` to then have all of your input fields as custom ones which are loaded in via functions based on the the use-case selected and the default of them pre-filled.
```
Comment (marinahand) [inline docs/_reference/use-cases/ai-app.md:59]: include a link to https://github.com/zapier/zapier-platform/blob/main/packages/schema/docs/build/schema.md#fieldschema

File: docs/_reference/use-cases/ai-app.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+**Pain Point:** Users often need help figuring out how to set up Zaps for common use cases.
+
+**Recommendation:**
+
+* Provide Zap Templates:  
+  * Create templates for common use cases with pre-mapped variables and good starter prompts.  
+  * **Why:** Templates serve as a starting point, helping users quickly set up their Zaps without having to configure everything from scratch.
+
+By addressing these pain points with the recommended strategies, you can significantly improve the user experience and functionality of AI apps on the Zapier platform. For detailed guidance and support, always refer to the [Zapier Developer Documentation](https://platform.zapier.com/docs) or contact the [Zapier support team](https://zapier.com/help/contact/).
```
Comment (marinahand) [inline docs/_reference/use-cases/ai-app.md:72]: modify to
https://developer.zapier.com/contact

File: docs/_reference/use-cases/ai-app.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+**Pain Point:** Actions that take a long time to process can cause issues, especially with Zapier's execution time limits. Test steps in the Zapier editor are limited to 50 seconds, and live Zap executions default to 45 seconds.
+
+**Recommendations:**
+
+* Use `z.generateCallbackUrl()` and `performResume`
+  * `z.generateCallbackUrl()`: This method generates a callback URL that your service can call once the long-running task is complete.  
+  * `performResume`: This function allows the action to pause until the callback URL is called, resuming once the task is complete.  
+  * **Why:** These tools help handle tasks that exceed Zapier's execution time limits by offloading the wait to an external service and resuming only when the task is done [Source].
+* Contact Zapier Partner Support:  
+  * If you need more time for live Zap executions, you can request an extension by contacting Zapier Partner Support through their [contact form](https://zapier.com/partner/contact/). However, we can only do this so much and the most foolproof way would be to implement the previous recommendation.
```
Comment (marinahand) [inline docs/_reference/use-cases/ai-app.md:25]: discussion in Slack thread if this is offered

Comment (mynamebrody) [reply docs/_reference/use-cases/ai-app.md]: Removed this based on Slack discussions

**Review by marinahand — APPROVED** [review]

**Pull request merged**